src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -215,9 +205,10 @@ private async ValueTask<CancellationTokenRegistration> HandleWriteStartState(Can
             });
 
             // Make sure start has completed
-            if (shouldAwaitStart)
+            if (!_started)
             {
-                await _startTask.ConfigureAwait(false);
+                await _sendResettableCompletionSource.GetTypelessValueTask().ConfigureAwait(false);
```

Comment:
I think we should guard better against attempted parallel writes. ManualResetValueTaskSourceCore will fail if you try to await it multiple times concurrently.

Also, why does this await have ConfigureAwait(false), but not other awaits?

Reply:
I'll file an issue. Also I don't think I need ConfigureAwait(false)

Reply:
https://github.com/dotnet/runtime/issues/32190